tools/lintapidiff.ml

```diff
@@ -125,20 +129,26 @@ module Ast = struct
       let self = add_item ~f path inherits in
       match ty.pmty_desc with
       | Pmty_signature lst -> List.fold_left self map lst
-      | Pmty_functor ({txt;_}, _, m) ->
-          let new_name = Ident.create_local txt in
+      | Pmty_functor ((Named ({txt = Some txt0},_)),m) ->
+          let new_name = Ident.create_local txt0 in
           let path = Path.Papply(path, Path.Pident new_name) in
           add_module_type path m (inherits, map)
       | Pmty_ident _ | Pmty_with _ | Pmty_typeof _| Pmty_extension _
-      | Pmty_alias _ -> map
+      | Pmty_alias _ | Pmty_functor (Named ({txt=None; _ }, _), _)
+      | Pmty_functor (Unit, _) -> map
     in
     let enter_path path name ty attrs map =
       let path = Path.Pdot (path, name.txt) in
       let inherits = f inherits name.loc attrs in
       add_module_type path ty (inherits, map)
     in
     let add_module map m =
-      enter_path  path m.pmd_name m.pmd_type m.pmd_attributes map
+      let name =
+        match m.pmd_name.txt with
+        | None -> { m.pmd_name with txt = "NONE"}
```

Comment:
I think it is better to fail with an `unsupported` here rather than trying to give an unique name to all unamed functor arguments.